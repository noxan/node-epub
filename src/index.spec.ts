import { EPub } from './index';

describe('node-epub', () => {
  it('should read epub metadata from file', () => {
    const epub = new EPub('./examples/lorem-ipsum-pages.epub');
    expect(epub.metadata).toMatchSnapshot();
  });
});
