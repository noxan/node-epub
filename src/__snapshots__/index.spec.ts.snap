// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`node-epub should read epub metadata from file 1`] = `
Object {
  "contributor": Object {
    "id": "contributor",
    "text": "Pages Publishing macOS v1.0",
  },
  "creator": Object {
    "id": "creator",
    "text": "Richard",
  },
  "date": "2020-02-12",
  "identifier": Object {
    "id": "BookId",
    "text": "345A4F7A-D930-4545-B1E4-3FAF38917550",
  },
  "language": "en",
  "meta": Array [
    Object {
      "property": "role",
      "refines": "#creator",
      "scheme": "marc:relators",
      "text": "aut",
    },
    Object {
      "content": "Pages Publishing macOS v1.0",
      "name": "generator",
    },
    Object {
      "property": "role",
      "refines": "#contributor",
      "scheme": "marc:relators",
      "text": "bkp",
    },
    Object {
      "property": "dcterms:modified",
      "text": "2020-02-12T11:01:47Z",
    },
    Object {
      "property": "ibooks:specified-fonts",
      "text": true,
    },
    Object {
      "property": "ibooks:version",
      "text": 1,
    },
    Object {
      "content": "cover-image",
      "name": "cover",
    },
  ],
  "subject": "Reference",
  "title": "Lorem ipsum",
}
`;
